internal/workload/wrapper.go

```diff
@@ -359,3 +364,27 @@ func getHostPorts(workload *v1.Pod) ([]int32, error) {
 	}
 	return hostPorts, nil
 }
+
+func isPortAvailable(port int32) bool {
+	conn, err := net.Listen("tcp", fmt.Sprintf("localhost:%d", port))
```

Comment:
If it will always be on `localhost` that the port availability needs to be tested, maybe name the function `isLocalPortAvailable` for clarity of the intent and readability. If you feel that `Local` it is redundant with the context of the execution in a container, it's fine by me

Reply:
Changed host address to be 0.0.0.0, that should make the context clear and also validate that the port is available across interfaces. What do you think now?

Reply:
looks great! :)